pkg/router/template/configmanager/haproxy/manager.go

```diff
@@ -1076,9 +1096,9 @@ func applyMapAssociations(m *HAProxyMap, associations map[string]string, add boo
 
 // backendModAnnotations return the annotations in a route that will
 // require custom (or modified) backend configuration in haproxy.
-func backendModAnnotations(route *routev1.Route) map[string]string {
+func backendModAnnotations(route *routev1.Route, cm *haproxyConfigManager) map[string]string {
 	termination := routeTerminationType(route)
-	backendModifiers := modAnnotationsList(termination)
+	backendModifiers := modAnnotationsList(termination, cm.listExtraAnnotations)
```

Comment:
rather than pass the `cm.{custom,listExtra}Annotations` to the function -- which is just appending to the returned list, you could just as well append it to `backendModifiers` ala:
`backendModifiers = append(backendModifiers, cm.customAnnotations...)`